File: net/src/SimpleMessageHeaderData.cpp
/*
 * SimpleMessageHeaderData.cpp
 *
 * Copyright (C) 2006 - 2010 by Visualisierungsinstitut Universitaet Stuttgart.
 * Alle Rechte vorbehalten.
 */

#include "vislib/SimpleMessageHeaderData.h"


#include <climits>


/*
 * vislib::net::VLSNP1_FIRST_RESERVED_MESSAGE_ID
 */
const vislib::net::SimpleMessageID vislib::net::VLSNP1_FIRST_RESERVED_MESSAGE_ID
    = static_cast<vislib::net::SimpleMessageID>(UINT_MAX - 1024);

File: extra/Vislib.Interop/SimpleMessageHeaderData.cs
/*
 * SimpleMessageHeaderData.cs
 *
 * Copyright (C) 2010 by Universitaet Stuttgart (VISUS). 
 * Alle Rechte vorbehalten.
 */

using System;
using System.Collections.Generic;
using System.Text;
using System.Runtime.InteropServices;


namespace Vislib.Interop {

    /// <summary>
    /// This is the message header that goes over the wire.
    /// </summary>
    [StructLayout(LayoutKind.Sequential)]
    public struct SimpleMessageHeaderData {

        /// <summary>
        /// User-defined message ID.
        /// </summary>
        public UInt32 MessageID;

        /// <summary>
        /// Size of the body to follow in bytes.
        /// </summary>
        public UInt32 BodySize;
    }

}
